src/views/virtualmachines/actions/components/VirtualMachineMigration/hooks/migrateVM.ts

```diff
@@ -0,0 +1,195 @@
+import { IoK8sApiCoreV1PersistentVolumeClaim } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { MAX_NAME_LENGTH } from '@kubevirt-utils/components/SSHSecretModal/utils/constants';
+import { getName, getNamespace } from '@kubevirt-utils/resources/shared';
+import { getDataVolumeTemplates, getVolumes } from '@kubevirt-utils/resources/vm';
+import { UPDATE_STRATEGIES } from '@kubevirt-utils/resources/vm/utils/constants';
+import { getStorageClassName } from '@kubevirt-utils/resources/vm/utils/dataVolumeTemplate/selectors';
+import { getRandomChars, isEmpty } from '@kubevirt-utils/utils/utils';
+import { k8sCreate, k8sDelete, k8sPatch, Patch } from '@openshift-console/dynamic-plugin-sdk';
+import { getMigrationClaimNameAnnotation } from '@virtualmachines/actions/utils';
+
+const getBlankDataVolume = (
+  dataVolumeName: string,
+  namespace: string,
+  storageClassName: string,
+  storage: string,
+): V1beta1DataVolume => {
+  const randomChars = getRandomChars();
+
+  const originName = dataVolumeName.replace(/-mig-[\d\w]+$/, '');
+  const namePrefix = `${originName}-mig`.substring(0, MAX_NAME_LENGTH - 5 - randomChars.length);
+
+  return {
+    apiVersion: `${DataVolumeModel.apiGroup}/${DataVolumeModel.apiVersion}`,
+    kind: DataVolumeModel.kind,
+    metadata: {
+      name: `${namePrefix}-${randomChars}`,
+      namespace,
+    },
+    spec: {
+      source: {
+        blank: {},
+      },
+      storage: {
+        resources: {
+          requests: {
+            storage,
+          },
+        },
+        storageClassName,
+      },
+    },
+  };
+};
+
+const createBlankDataVolumes = async (
+  selectedPVCs: IoK8sApiCoreV1PersistentVolumeClaim[],
+  destinationStorageClass: string,
+): Promise<Map<string, V1beta1DataVolume>> => {
+  const dataVolumesCreationResults = await Promise.allSettled(
+    selectedPVCs
+      .map((pvc) => {
+        return getBlankDataVolume(
+          getName(pvc),
+          getNamespace(pvc),
+          destinationStorageClass,
+          pvc?.spec?.resources?.requests?.storage,
+        );
+      })
+      .map((dataVolume) =>
+        k8sCreate({ data: dataVolume, model: DataVolumeModel, ns: getNamespace(dataVolume) }),
+      ),
+  );
+
+  const { fulfilledRequests, rejectedRequests } = dataVolumesCreationResults.reduce(
+    (acc, result) => {
+      result.status === 'rejected'
+        ? acc.rejectedRequests.push(result)
+        : acc.fulfilledRequests.push(result);
+
+      return acc;
+    },
+    {
+      fulfilledRequests: [] as PromiseFulfilledResult<V1beta1DataVolume>[],
+      rejectedRequests: [] as PromiseRejectedResult[],
+    },
+  );
+
+  if (!isEmpty(rejectedRequests)) {
+    fulfilledRequests.map((result) =>
+      k8sDelete({
+        model: DataVolumeModel,
+        resource: result.value,
+      }),
+    );
+
+    throw new Error(rejectedRequests?.[0].reason);
```

Comment:
?

Reply:
This is the single vm migration. When we create data volumes but something fails, just throw an exception and delete the created datavolumes.